x/distribution/keeper/genesis.go

```diff
@@ -97,12 +97,11 @@ func (k Keeper) InitGenesis(ctx sdk.Context, data types.GenesisState) {
 
 	balances := k.bankKeeper.GetAllBalances(ctx, moduleAcc.GetAddress())
 	if balances.IsZero() {
-		if err := k.bankKeeper.SetBalances(ctx, moduleAcc.GetAddress(), moduleHoldingsInt); err != nil {
-			panic(err)
-		}
-
 		k.authKeeper.SetModuleAccount(ctx, moduleAcc)
 	}
+	if !balances.IsEqual(moduleHoldingsInt) {
+		panic(fmt.Sprintf("distribution module balance does not match the module holdings: %s <-> %s", balances, moduleHoldingsInt))
+	}
```

Comment:
we are changing a behavior here. For example: in the original code, when `balances.IsZero() == true` then we are still in a happy path - we add set the expected balance. Now, we don't update the balance and throw an error instead.

Reply:
it's the correct approach, after a case study this behaviour was abused in tests only. we should fail if a module thinks it holds coins while bank does not reflect that. it might mean only one thing: corrupted or malformed genesis file.